src/text/text.c

```diff
@@ -103,12 +139,22 @@ void n_graphics_draw_text_ex(
 
     uint32_t codepoint = 0, next_codepoint = 0, last_codepoint = 0,
         last_renderable_codepoint = 0, last_breakable_codepoint = 0;
+
+    // Initialize cache for glyph widths
+    int8_t * glyph_widths = malloc(strlen(text) * sizeof(int8_t));
```

Comment:
I'm not a huge fan of allocating here, especially because this function returns `void` and has no way to propagate an error if allocation fails.  But we should at least not crash if allocation fails, so `if (!glyph_widths) return;` is probably in order here.  (Also, I think that inside Neographics, we use `NGFX_PREFERRED_malloc` and `NGFX_PREFERRED_free`.)

I don't entirely understand at a skim how many times the cache gets hit.  How much extra pain do we go through if we just look up a new glyph each time, rather than caching?

Reply:
My bad - I completely agree with what you've pointed it out. I somehow (wrongly) assumed that `n_graphics_font_get_glyph_info` was more expensive then it actually was now that I dug into that function. I'll adjust the code to not bother with the cache.